### ISSUE TITLE
2K2K dataset testset obj file missing Col attribute

### ISSUE THREAD

**Issue by zzg-zzg** [SangHunHan92/2K2K#35]
I previously requested access to the older version of the 2K2K dataset last year. Recently, while finalizing my paper and rendering the test set (50 samples), I noticed that only the first obj file (ID: 07184) contains the color attribute (Col), while the subsequent 49 obj files appear to lack this attribute. Today, I have resubmitted an application via HuggingFace to request dataset access permissions.

However, my primary concern is whether you are aware of this Col attribute absence in the majority of test set obj files (excluding the first one). I kindly request that you verify the integrity of the latest dataset version and, if confirmed, implement necessary corrections to resolve this issue.

**Comment by jseobyun** [comment]
Hi, @zzg-zzg , this is jseob.y that processes new version of [Polygom2K2K](https://huggingface.co/datasets/polygom-team/2K2K).

Firstly, I'm sorry for giving you confusion. Could you explain more detail about your argument? 

For example, which tool do you use or loading code (e.g. trimesh, pytorch3d, open3d ,etc...)

In my case, all test sample of Polygom2K2K 5000 ply files contain colors as follow :

![Image](https://github.com/user-attachments/assets/e615571f-21b0-44e9-aaea-2bf1e6d0d18c)

I used trimesh.load and the sample contains mesh.visual.face_colors and mesh.visual.vertex_color correctly. 

---------
NOTE

The old version of 2K2K you downloaded is unreachable for me, so if you need more discussion with me. 

Please download Polygom2K2K (new version) and leave issues on HuggingFace.

I will happy to help you :)

**Comment by zzg-zzg** [comment]
Thank you for your reply. I use the old version 2k2k downloaded from Google Drive. The test set is named 100k and has 50 OBJ files. I use Blender to render all the train and val data under '1M' folder. The OBJ in '1M' folder works well. 
I also resubmitted a request using Google form and got this.  

![Image](https://github.com/user-attachments/assets/b481e5ec-a6f3-4800-a1d9-04e6fbdaa658)
Please let me know if anything I need to do after that.

**Comment by zzg-zzg** [comment]
I downloaded the new version and found it still misses the 'Col' attribute in the test set. ![Image](https://github.com/user-attachments/assets/3dbbca56-0d7e-4502-a6fa-3d8cd388376f). The old version of the 2k2k OBJ file in the train set has a Col attribute 

![Image](https://github.com/user-attachments/assets/e2812c06-48e0-4abc-8a59-5791134cbeac), but missing in test set OBJs

![Image](https://github.com/user-attachments/assets/61e7b965-1480-4fef-8104-ca7208209996). I also find the two version testset only different with '07184' OBJ. How can I get color in Blender now?

**Comment by zzg-zzg** [comment]
I wrote a script to handle this problem by transferring the .ply to .obj, and the attribute now can be loaded by Blender. 
`import os
import trimesh
from pathlib import Path

def save_as_obj_with_color(mesh, output_path):
    """Save the Trimesh object as an .obj file with color"""
    mesh.export(output_path, file_type='obj')

def convert_ply_to_obj(ply_path, obj_output_path):
    """Convert the PLY file to an .obj file and preserve color"""
    mesh = trimesh.load(ply_path)  # Load the PLY file
    save_as_obj_with_color(mesh, obj_output_path)  # Save it as an .obj file with color

def process_folders(base_dir, output_base_dir):
    """Iterate through each folder in base_dir, process the PLY file in each folder, and save it as OBJ"""
    # Get all subfolders in base_dir
    folder_names = os.listdir(base_dir)
    for folder_name in folder_names:
        folder_path = os.path.join(base_dir, folder_name)

        # If it's a directory and contains a .ply file
        if os.path.isdir(folder_path):
            ply_file_path = os.path.join(folder_path, f"{folder_name}.ply")
            if os.path.exists(ply_file_path):
                # Create the target folder with the same name as the original folder
                output_folder_path = os.path.join(output_base_dir, folder_name)
                Path(output_folder_path).mkdir(parents=True, exist_ok=True)

                # Define the path to save the .obj file
                obj_output_path = os.path.join(output_folder_path, f"{folder_name}.obj")

                # Convert the PLY file to OBJ and save it
                convert_ply_to_obj(ply_file_path, obj_output_path)
                print(f"Successfully converted {ply_file_path} to {obj_output_path}")
            else:
                print(f"Skipped folder {folder_name}: {folder_name}.ply not found")

if __name__ == "__main__":
    base_directory = '/data/zzg/split_1'  # Input folder path
    output_directory = '/data/zzg/split_obj'  # Output folder path

    # Process the folders
    process_folders(base_directory, output_directory)
`
